src/SfxWeb/src/app/Models/DataModels/collections/CollectionBase.ts

```diff
@@ -76,16 +77,19 @@ export class DataModelCollectionBase<T extends IDataModel<any>> implements IData
         let success = true;
         if (!this.refreshingPromise) {
             this.refreshingPromise = new AsyncSubject<any>();
+            this.clearOnFailureToUpdate = true;
 
             this.retrieveNewCollection(messageHandler).pipe(mergeMap(collection => {
                 return this.update(collection);
             }),
             catchError( err => {
-                this.collection = [];
                 success = false;
                 return of(err);
             })
             ).subscribe( () => {
+                if(this.clearOnFailureToUpdate && this.lastRefreshWasSuccessful){
```

Comment:
We only have to clear the collection if the previous data is valid (last refresh had no problem) and the current one failed. Decided to put it in the subscribe function because it makes more sense to check it once it is done.